### ISSUE TITLE
'accountName' exceeds maximum length of '24'

### ISSUE THREAD

**Issue by ealsur** [Azure/azure-sdk-for-net#9848]
**Describe the bug**
Using the Management APIs `Microsoft.Azure.Management.Storage.Fluent` I cannot call `IStorageAccounts.GetByResourceGroupAsync` with an account name longer than 24 characters although that account name already exists and I can see it if I use `ListAsync`.

**Expected behavior**
Support existing accounts.

**Actual behavior (include Exception or Stack Trace)**

```csharp
Microsoft.Rest.ValidationException: 'accountName' exceeds maximum length of '24'.
   at Microsoft.Azure.Management.Storage.Fluent.StorageAccountsOperations.GetPropertiesWithHttpMessagesAsync(String resourceGroupName, String accountName, Nullable`1 expand, Dictionary`2 customHeaders, CancellationToken cancellationToken)     at Microsoft.Azure.Management.Storage.Fluent.StorageAccountsOperationsExtensions.GetPropertiesAsync(IStorageAccountsOperations operations, String resourceGroupName, String accountName, Nullable`1 expand, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.Storage.Fluent.StorageAccountsImpl.GetInnerByGroupAsync(String groupName, String name, CancellationToken cancellationToken)
   at Microsoft.Azure.Management.ResourceManager.Fluent.Core.GroupableResources`5.GetByResourceGroupAsync(String groupName, String name, CancellationToken cancellationToken)
```

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

- Create a Storage Account on the Azure Portal with a name larger than 24 characters, this is currently possible.

- Try to read it using `GetByResourceGroupAsync`:

```csharp
    AzureCredentials credentials = SdkContext.AzureCredentialsFactory.FromServicePrincipal(...);

    Microsoft.Azure.Management.Fluent.Azure.IAuthenticated authenticatedAzure = Microsoft.Azure.Management.Fluent.Azure
        .Configure()
        .Authenticate(credentials);

    Microsoft.Azure.Management.Fluent.IAzure iAzure = authenticatedAzure
        .WithSubscription(....);

    var response = await iAzure.StorageAccounts.GetByResourceGroupAsync(
        "resourceGroupName",
        "myaccountnamelongerthan24")
```

**Environment:**
 - Microsoft.Azure.Management.Storage.Fluent 1.30.0
 - Windows
 - dotnet SDK 3.1 (netcoreapp3.1)

**Comment by jsquire** [comment]
//cc: @yaohaizh

**Comment by yungezz** [comment]
hi @xccc-msft could you pls help to confirm there's no constraint at fluent layer? thanks.

**Comment by xccc-msft** [comment]
@ealsur The length restriction was defined by service team in the [swagger](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2018-11-01/storage.json#L2343-L2344). 

This could also be found via [storage .NET SDK](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Microsoft.Azure.Management.Storage/src/Generated/StorageAccountsOperations.cs#L275-L278).

We have added the label of `Service Attention`. They would help check and make required changes in swagger. After that, we could plan to regenerate the code.

**Comment by amishra-dev** [comment]
@ealsur account name limit is 24 characters from the service. here is the doc that says the same https://docs.microsoft.com/en-us/azure/storage/common/storage-account-overview#naming-storage-accounts

**Comment by ealsur** [comment]
@amishra-dev I was able to create accounts with larger names, the failure we were getting was trying to access those accounts programmatically. Looks like the limit on the Portal was added later, as these are accounts we already have.

Below is a screenshot from the Azure Portal, the name `cosmosdbjobschedulertest` has 26 characters.

![image](https://user-images.githubusercontent.com/1633401/92967398-0a7f2880-f42e-11ea-8f01-4b906a2d4725.png)
